includes/post-type-functions.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Post type functions.
+ *
+ * @link       https://nataliesmith.dev
+ * @since      1.0.0
+ * @package    Poetwp
+ * @subpackage Poetwp/includes
+ */
+
+/**
+ * Register the custom post type
+ */
+function poetwp_register_post_type() {
+	register_post_type(
+		'poem',
+		array(
+			'labels'       => array(
+				'name'          => __( 'Poems' ),
+				'singular_name' => __( 'Poem' ),
+			),
+			'public'       => true,
+			'has_archive'  => true,
+			'supports'     => array( 'title', 'editor', 'thumbnail', 'author', 'excerpt', 'revisions' ),
+			'menu_icon'    => 'dashicons-format-aside',
+			'rewrite'      => array(
+				'slug' => 'poems',
+			),
+			'show_in_rest' => true,
+		)
+	);
+}
+// Hook the registration function to the 'init' action.
+add_action( 'init', 'poetwp_register_post_type' );
+
+/**
+ * Hide the default title for poem post type.
```

Comment:
Why?

I can see that the title is filtered into the_content in the next function. My concern would be how themes might interpret this - maybe they've got theme markup that has a `header` element and a `div`, how that might end up with an empty header element, and what assistive technologies (for accessibility) might make of that.

Reply:
This is because one of the requirements was for the custom fields (date and notes) to go before the title. I tested 3 different themes and struggled to find a method that worked across all of them to achieve this. Different themes provide thier own hooks to handle the header element, and block themes seem to use block groups for the header. So this was the only solution that kind of worked for all of them (out of the box, at least)